sources/api/netdog/src/cli/check_net_config.rs

```diff
@@ -0,0 +1,38 @@
+use super::{check_net_config, error, Result};
+use argh::FromArgs;
+use snafu::OptionExt;
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand, name = "check-net-config")]
+/// Check the network configuration
+pub(crate) struct CheckNetConfigArgs {}
+
+/// Check that the configuration for network interfaces parses
+pub(crate) fn run() -> Result<()> {
+    // `maybe_net_config` could be `None` if no interfaces were defined
+    let net_config = match check_net_config() {
+        Ok(Some(net_config)) => net_config,
+        Ok(None) => {
+            eprintln!("No network interfaces were configured");
+            return Ok(());
```

Comment:
Shouldn't the process exit non-zero if the configuration is invalid? i.e. shouldn't this return an actual error?

Reply:
Thats a good question. For `generate-net-config` its considered desirable to exit 0 on this case. Technically it did parse correctly, but it would not result in useful network configuration (which my update to the docs says is technically right). I guess we could envision this being added to a workflow to validate the configuration which would mean we should be very precise about exit codes.

Reply:
In the case that this branch is hit, the following would be printed:
```bash
No network interfaces were configured
```
with a zero exit. The file *did* parse, but it probably doesn't provide meaningful network configuration. I think this is still correct behavior since its just validating that it parses, not that it will work for a particular setup.